storage-api/src/java/org/apache/hive/common/util/BloomKFilter.java

```diff
@@ -527,20 +527,19 @@ public void add(ElementWrapper wrapper) {
     @Override
     public void run() {
       while (!executor.isTerminated() && !queue.isEmpty()) {
-        ElementWrapper currentBf = queue.poll();
+        ElementWrapper currentBf = null;
+        try {
+          currentBf = queue.take();
+        } catch (InterruptedException e) {
```

Comment:
Do not ignore.  An Interrupt means that it's time to exit.

Reply:
right, I'll do a simple method return with an LOG.warn to have some trace in the logs